backend/app/api/api_v1/routers/tests/test_search.py

```diff
@@ -0,0 +1,171 @@
+    response3_json = response3.json()
+    del response3_json["query_time_ms"]
+
+    assert response1_json["documents"]
+    assert response1_json == response2_json == response3_json
+
+
+def test_punctuation_ignored(test_opensearch, monkeypatch, client, user_token_headers):
+    monkeypatch.setattr(search, "_OPENSEARCH_CONNECTION", test_opensearch)
+
+    response1 = client.post(
+        "/api/v1/searches",
+        json={"query_string": "disaster.", "exact_match": False},
+        headers=user_token_headers,
+    )
+    response2 = client.post(
+        "/api/v1/searches",
+        json={"query_string": "disaster, ", "exact_match": False},
+        headers=user_token_headers,
+    )
+    response3 = client.post(
+        "/api/v1/searches",
+        json={"query_string": ";disaster", "exact_match": False},
+        headers=user_token_headers,
+    )
+
+    response1_json = response1.json()
+    del response1_json["query_time_ms"]
+    response2_json = response2.json()
+    del response2_json["query_time_ms"]
+    response3_json = response3.json()
+    del response3_json["query_time_ms"]
+
+    assert response1_json["documents"]
+    assert response1_json == response2_json == response3_json
+
+
+def test_accents_ignored(test_opensearch, monkeypatch, client, user_token_headers):
+    monkeypatch.setattr(search, "_OPENSEARCH_CONNECTION", test_opensearch)
+
+    response1 = client.post(
+        "/api/v1/searches",
+        json={"query_string": "disàster", "exact_match": False},
+        headers=user_token_headers,
+    )
+    response2 = client.post(
+        "/api/v1/searches",
+        json={"query_string": "disastër", "exact_match": False},
+        headers=user_token_headers,
+    )
+    response3 = client.post(
+        "/api/v1/searches",
+        json={"query_string": "disàstér", "exact_match": False},
+        headers=user_token_headers,
+    )
+
+    response1_json = response1.json()
+    del response1_json["query_time_ms"]
+    response2_json = response2.json()
+    del response2_json["query_time_ms"]
+    response3_json = response3.json()
+    del response3_json["query_time_ms"]
+
+    assert response1_json["documents"]
+    assert response1_json == response2_json == response3_json
+
+
+def test_unauthenticated(client):
+    response = client.post(
+        "/api/v1/searches",
+        json={"query_string": "a", "exact_match": True},
+    )
+    assert response.status_code == 401
+
+
+def test_time_taken(test_opensearch, monkeypatch, client, user_token_headers):
+    monkeypatch.setattr(search, "_OPENSEARCH_CONNECTION", test_opensearch)
+
+    start = time.time()
+    response = client.post(
+        "/api/v1/searches",
+        json={"query_string": "disaster", "exact_match": False},
+        headers=user_token_headers,
+    )
+    end = time.time()
+
+    processed_response_time_ms = 1000 * (end - start)
+    assert response.status_code == 200
+    assert 0 < response.json()["query_time_ms"] < processed_response_time_ms
```

Comment:
Is it more a case of checking this attribute is in the response, rather than checking rough accuracy?
Or perhaps, if you have a "really fast computer" :-)

```
assert 0 < response.json()["query_time_ms"] <= processed_response_time_ms
```

?

Reply:
Some of the responses I've seen have been in the low single digit ms, so while this isn't a great test,,, it did at least check that some time had passed and it wasn't obviously wrong... I'll think about how to make it better!